internal/daemon/worker/testing.go

```diff
@@ -548,16 +539,62 @@ func NewTestMultihopWorkers(t testing.TB,
 		WorkerAuthDebuggingEnabled: enableAuthDebugging,
 		DisableAutoStart:           true,
 	})
+	t.Cleanup(childKmsWorker.Shutdown)
 	childKmsWorker.w.conf.WorkerAuthStorageKms = nil
 
 	err = childKmsWorker.w.Start()
-	t.Cleanup(childKmsWorker.Shutdown)
-	if err != nil {
-		t.Fatal(err)
-	}
+	require.NoError(err)
 
-	// Sleep so that workers can startup and connect.
-	time.Sleep(12 * time.Second)
+	t.Log("Waiting for workers to start up")
+	require.Eventually(
+		func() bool {
+			t.Log("Checking worker status")
+			workers, err := serversRepo.ListWorkers(controllerContext, []string{"global"})
```

Comment:
Question: Is `NewTestMultihopWorkers` ever called by parallel tests? Could this impact the output of ListWorkers?

Reply:
Not as long as each test is using it's own controller, which I think we have to assume the parallel tests are doing.